### PR TITLE
Adds brokens links section to edit steps page.

### PR DESCRIPTION
**Pull request by DilwoarH**
It uses a panel and a table to display the information regarding the broken links. A table was used as there was a lot of data which is not easily comprehended via other methods. Table seems to be the most sensible way of doing things.

Ticket: https://trello.com/c/EPGwXv7z/768-add-styling-to-link-report-information-on-edit-steps-page-s

screen shot 2018-08-09 at 12 46 47

### CODE REVIEW

File: app/views/steps/_form.html.erb [hunk 1]
```diff
@@ -59,10 +59,55 @@
...
+                <tr>
+                  <td><%= link['uri'] %></td>
+                  <td><%= link['status'] %></td>
+                  <td><%= link['checked'] %></td>
+                  <td><%= link['problem_summary'] %></td>
+                  <td>
+                    <ul>
+                    <% link['errors'].each do |error| %>
+                      <li><%= error %></li>
+                    <% end %>
```
Comment (andysellick) [inline app/views/steps/_form.html.erb]: Nitpick: indentation.

File: app/views/steps/_form.html.erb [hunk 2]
```diff
@@ -59,10 +59,55 @@
...
+            <thead>
+            <tbody>
+              <% step.broken_links.each do |link| %>
+                <tr>
+                  <td><%= link['uri'] %></td>
+                  <td><%= link['status'] %></td>
+                  <td><%= link['checked'] %></td>
+                  <td><%= link['problem_summary'] %></td>
+                  <td>
+                    <ul>
```
Comment (andysellick) [inline app/views/steps/_form.html.erb]: This isn't your fault, but the standard bootstrap list looks like it wastes a lot of space on the left. Don't suppose there's a way of compressing that, or displaying this information in an alternative way?

File: app/views/steps/_form.html.erb [hunk 3]
```diff
@@ -59,10 +59,55 @@
...
+                    <% link['errors'].each do |error| %>
```
Comment (andysellick) [inline app/views/steps/_form.html.erb]: I'm just curious - does this ever output more than one error? Presumably a broken link can only return one status code, whether that's 404 or 500 or whatever.

File: app/views/steps/_form.html.erb [hunk 4]
```diff
@@ -59,10 +59,55 @@
...
+                      <li><%= error %></li>
+                    <% end %>
+                    </ul>
+                  </td>
+                  <td>
+                    <ul>
+                    <% link['warnings'].each do |warning| %>
```
Comment (andysellick) [inline app/views/steps/_form.html.erb]: What warnings can we get from a broken link? i.e. is this column of the table needed?

File: app/views/steps/_form.html.erb [hunk 5]
```diff
@@ -59,10 +59,55 @@
...
+                      <li><%= warning %></li>
+                    <% end %>
+                    </ul>
+                  </td>
+                  <td><%= link['suggested_fix'] %></td>
```
Comment (andysellick) [inline app/views/steps/_form.html.erb]: What are the possible suggested fixes? If there's more than one, fine, but if they're all going to suggest the same thing (find a link that isn't broken) this column feels a bit redundant.

**Comment by DilwoarH** [comment]
Closing as I rebased with master and feature branch is outdated

**Comment by DilwoarH** [comment]
NEW PR https://github.com/alphagov/collections-publisher/pull/450/files

**Pull request closed without merging**